### PR TITLE
Read translated custom property name

### PR DESCRIPTION
**Pull request by dantleech**
Follow up to: https://github.com/doctrine/phpcr-odm/pull/602

I wonder what else I missed, maybe "assoc" fields. Will add some better tests in this PR.

### CODE REVIEW

**Comment by dbu** [comment]
ups. maybe search the codebase for `$mapping['property']`

**Comment by dantleech** [comment]
null fields are also still named after the field names and not the property names

**Comment by dantleech** [comment]
ok, fixed the null fields and a few other things.

File: tests/Doctrine/Tests/ODM/PHPCR/Functional/Translation/AttributeTranslationStrategyTest.php [hunk 1]
```diff
@@ -167,13 +171,20 @@ public function testTranslationNullProperties()
...
         $strategy->loadTranslation($doc, $node, $this->metadata, 'fr');
         $this->assertEquals('Un sujet intéressant', $doc->topic);
         $this->assertEquals('Lorem français', $doc->getText());
         $this->assertNull($doc->nullable);
+        $this->assertNull($doc->customPropertyName);
         $this->assertEquals(array(), $doc->getSettings());
+
+        $nullFields = $node->getProperty('phpcr_locale:fr' . AttributeTranslationStrategy::NULLFIELDS)->getValue();
+        $this->assertEquals(array(
+            'custom-property-name',
```
Comment (dantleech) [inline tests/Doctrine/Tests/ODM/PHPCR/Functional/Translation/AttributeTranslationStrategyTest.php]: Why do we store the nullfields of scalar values here? Would not the abscence of any field make it implicitly null?

Comment (dbu) [inline tests/Doctrine/Tests/ODM/PHPCR/Functional/Translation/AttributeTranslationStrategyTest.php]: the problem is when all translated fields are allowed to be null, you don't know if a "translation" in that language exists or not, i think. uwe worked on that, maybe he remembers more exactly.

**Comment by dbu** [comment]
thanks! i merged, it certainly is more correct. if you find more issues, please tell.

hm, btw, was setting the property name manually completely ignored before or did it not work at all? if it was ignored, we should add a note to the changelog, as people who had a property name on translated fields will need to adjust their content.

File: tests/Doctrine/Tests/ODM/PHPCR/Functional/Translation/AttributeTranslationStrategyTest.php (hunk 1 above)
Comment (dantleech) [inline tests/Doctrine/Tests/ODM/PHPCR/Functional/Translation/AttributeTranslationStrategyTest.php]: would it not be sufficient to have an extra single multivalue field which lists the active translations? I guess its done this way beacause of merging ? /cc @uwej711

**Comment by dantleech** [comment]
it was completely ignored by the attribute translation strategy, so yeah, if people did map `property` and didn't realise it didn't work they will have broken things.

**Comment by dbu** [comment]
ok, thanks for #606 - i think we still should do it. its one of those bugfixes that has a slight chance of breaking peoples code but its only going in the next minor version, not in a dot release.

**Pull request merged**